## Prueba #0201 

In [5]:
import pandas as pd
import os

#### Leer los datos.

In [6]:
# Ir al directorio en el que se encuentra el archivo. 
os.chdir(r'./')                # Reemplazar por la ruta correcta.

In [7]:
df = pd.read_csv('data_studentlist.csv', header='infer')

In [8]:
df.head(10)

name gender  age  grade absence bloodtype  height  weight
0     Jared Diamond      M   23      3       Y         O   165.3    68.2
1    Sarah O'Donnel      F   22      2       N        AB   170.1    53.0
2      Brian Martin      M   24      4       N         B   175.0    80.1
3      David Hassel      M   23      3       N        AB   182.1    85.7
4   Clara Rodriquez      F   20      1       Y         A   168.0    49.5
5  Jennifer Lorentz      F   21      2       N         O   162.0    52.0
6       Susan Clark      F   22      1       N         O   155.2    45.3
7   Margareth Jones      F   23      1       N         A   176.9    55.0
8      John Bertsch      M   23      3       N         B   178.5    64.2
9    Jake Timmerman      M   22      2       N         B   176.1    61.3

#### Responda a las preguntas utilizando la librería Pandas.

1). Altura media de los alumnos:

In [14]:
#Forma 1
df.loc[df['gender'] == 'M', ['height']].mean()

height    172.41
dtype: float64

In [26]:
#df = pd.DataFrame(lists, columns = ['name', 'gender', 'age', 'grade','absence', 'bloodtype', 'height', 'weight'])
#Forma 2
fil = df[df['gender'] == 'M']
fil['height'].mean()

172.41

2). Altura media de las alumnas:

In [15]:
#Forma 1
df.loc[df['gender'] == 'F', ['height']].mean()

height    166.642857
dtype: float64

In [13]:
#Forma 2
fil = df[df['gender'] == 'F']
fil['height'].mean()

166.64285714285714

3). Peso medio de los alumnos:

In [16]:
#Forma 1
df.loc[df['gender'] == 'M', ['weight']].mean()

weight    68.5
dtype: float64

In [18]:
#Forma 2
fil = df[df['gender'] == 'M']
fil['weight'].mean()

68.50000000000001

4). Peso medio de las alumnas:

In [19]:
#Forma 1
df.loc[df['gender'] == 'F', ['weight']].mean()

weight    50.442857
dtype: float64

In [20]:
#Forma 2
fil = df[df['gender'] == 'F']
fil['weight'].mean()

50.44285714285714

5). La altura más alta entre los alumnos:

In [21]:
#Forma 1
df.loc[df['gender'] == 'M', ['height']].max()

height    182.1
dtype: float64

In [22]:
#Forma 2
fil = df[df['gender'] == 'M']
fil['height'].max()

182.1

6). La altura más baja entre las alumnas:

In [23]:
#Forma 1
df.loc[df['gender'] == 'F', ['height']].max()

height    176.9
dtype: float64

In [24]:
#Forma 2
fil = df[df['gender'] == 'F']
fil['height'].max()

176.9

7). El peso menor entre los alumnos de estatura superior a 175 cm:

In [25]:
#Forma 1
df.loc[(df['gender'] == 'M') & (df['height'] > 175.00), ['weight']].min()

weight    61.3
dtype: float64

In [26]:
#Forma 2
fil = df[(df['gender'] == 'M') & (df['height'] > 175.00)]
fil['weight'].min()

61.3

8). El peso mayor entre las alumnas de altura inferior a 160 cm:

In [27]:
#Forma 1
df.loc[(df['gender'] == 'F') & (df['height'] < 160.00), ['weight']].max()

weight    45.3
dtype: float64

In [28]:
#Forma 2
fil = df[(df['gender'] == 'F') & (df['height'] < 160.00)]
fil['weight'].max()

45.3

9). La calificación promedio de los estudiantes sin ausencia ('N'):

In [29]:
#Forma 1
df.loc[df['absence'] == 'N', ['grade']].mean()

grade    2.076923
dtype: float64

In [30]:
#Forma 2
fil = df[df['absence'] == 'N']
fil['grade'].mean()

2.076923076923077

10). La calificación promedio de los estudiantes con ausencia ('Y'):

In [31]:
#Forma 1
df.loc[df['absence'] == 'Y', ['grade']].mean()

grade    3.0
dtype: float64

In [32]:
#Forma 2
fil = df[df['absence'] == 'Y']
fil['grade'].mean()

3.0

11). Altura media de los estudiantes con tipo de sangre 'A' o 'AB':

In [33]:
#Forma 1
df.loc[(df['bloodtype'] == 'A') | (df['bloodtype'] == 'AB'), ['height']].mean()

height    172.642857
dtype: float64

In [34]:
#Forma 2
fil = df[(df['bloodtype'] == 'A') | (df['bloodtype'] == 'AB')]
fil['height'].mean()

172.64285714285714

12). Altura promedio de los estudiantes **masculinos** con tipo de sangre 'A' o 'AB':

In [35]:
#Forma 1
df.loc[((df['bloodtype'] == 'A') | (df['bloodtype'] == 'AB')) & (df['gender'] == 'M'), ['height']].mean()

height    173.375
dtype: float64

In [23]:
#Forma 2
fil = df[((df['bloodtype'] == 'A') | (df['bloodtype'] == 'AB')) & (df['gender'] == 'M')]
fil['height'].mean()

173.375

13). Edad promedio de los estudiantes con ausencia ('Y') cuya calificación es igual o mayor a 3:

In [37]:
#Forma 1
df.loc[(df['absence'] == 'Y') & (df['grade'] >= 3), ['age']].mean()

age    23.666667
dtype: float64

In [38]:
#Forma 2
fil = df[(df['absence'] == 'Y') & (df['grade'] >= 3)]
fil['age'].mean()

23.666666666666668